pkg/resolver/searchHelper.go

```diff
@@ -459,3 +462,86 @@ func matchOperatorToProperty(dataType string, opValueMap map[string][]string,
 	}
 	return opValueMap
 }
+
+// partialMatchStringPattern checks if config.Cfg.HubName partially matches any pattern in the values slice.
+// It loops through each pattern, prepares it for matching, and checks for a match.
+// If a match is found, it returns true, indicating a match is found. Else, returns false.
+func partialMatchStringPattern(values []string) (bool, error) {
+	for _, pattern := range values {
+		klog.V(5).Info("ManagedHub filter pattern to match: ", pattern, " hubname: ", config.Cfg.HubName)
+		//fix prefix
+		if !strings.HasPrefix(pattern, "*") {
+			pattern = "^" + pattern
+		}
+		//fix suffix
+		if !strings.HasSuffix(pattern, "*") {
+			pattern = pattern + "$"
+		}
+		// fix start and end of string and match multiple characters
+		pattern = strings.ReplaceAll(pattern, "%", ".*")
+		matched, err := regexp.MatchString(pattern, config.Cfg.HubName)
+
+		if err != nil {
+			klog.Error("Error in partialMatchStringPattern for managedHub filter:", err)
+			return false, err
+		}
+		if matched {
+			return matched, nil
+		}
+	}
+	klog.V(4).Infof("%s not in managedHub filter %+v", config.Cfg.HubName, values)
+	return false, nil
+}
+
+// processOpValueMapManagedHub processes the key-value pair for a managedHub filter.
+// It handles different key cases such as "!", "!=", "=", "!:*", "!=:*", and "=:*".
+// It returns a boolean indicating whether the search should proceed based on the evaluation of the key and values.
+func processOpValueMapManagedHub(key string, values []string) bool {
```

Comment:
**Suggestion**: We can reduce the complexity moving the log to the end and including the inputs and results like this.

```
func processOpValueMapManagedHub(key string, values []string) bool {
	result := false
	switch key {
	case "!", "!=":
		result = !slices.Contains(values, config.Cfg.HubName)
	case "=":
		result = slices.Contains(values, config.Cfg.HubName)
	case "!:*", "!=:*":
		match, err := partialMatchStringPattern(values)
		if err != nil {
			klog.Errorf("Error processing partial match for ManagedHub filter:", err)
			return false
		}
		result = !match // Return the inverse of match to indicate search should not proceed if there is a partial match
	case "=:*":
		match, err := partialMatchStringPattern(values)
		if err != nil {
			klog.Errorf("Error processing partial match for ManagedHub filter:", err)
			return false
		}
		return match // Return match to indicate search should proceed if there is a partial match
	}

	klog.V(4).Infof("ManagedHub filter hubname: %s operation: %s values: %+v  result: %t",
	config.Cfg.HubName, key, values, result)

	return result
}
```